server/controllers/shoppingList.js

```diff
@@ -196,35 +197,47 @@ const addProductToList = (req, resp) => {
 };
 
 const getListData = (req, resp) => {
-  ShoppingList.find(
-    {
-      _id: req.params.id,
-      $or: [
-        { adminIds: req.user._id },
-        { ordererIds: req.user._id },
-        { purchaserIds: req.user._id }
-      ]
-    },
-    (err, documents) => {
-      if (err) {
-        return resp.status(404).send({ message: err.message });
+  let list;
+  ShoppingList.findOne({
+    _id: req.params.id,
+    $or: [
+      { adminIds: req.user._id },
+      { ordererIds: req.user._id },
+      { purchaserIds: req.user._id }
+    ]
+  })
+    .then(doc => {
+      if (!doc) {
+        return resp.status(404).send({ message: 'No list of given id!' });
       }
 
-      if (documents && documents.length > 0) {
-        const { cohortId, _id, isArchived } = documents[0];
+      list = doc;
 
-        if (isArchived) {
-          return resp.status(200).json({ cohortId, _id, isArchived });
-        }
+      const { cohortId } = list;
+      if (cohortId) {
+        return Cohort.findOne({ _id: cohortId }).then(cohort => {
```

Comment:
what about error checking here?

Reply:
I used 'thens' instead of callbacks (err, doc) => {} because there are other async queries to the db in those controllers. All errors are captured by the catch method at the end of thens chain.